### ISSUE TITLE
Do not require textbox role in combobox if input[type=search] exists

### ISSUE THREAD

**Issue by ldanet** [dequelabs/axe-core#1688]
This is effectively a duplicate of https://github.com/dequelabs/axe-core/issues/549. That issue is marked as closed but the issue persists.

Expectation: 
The code below should pass.
It is a copy paste of Example 1 of [ARIA 1.1 Combobox with Listbox Popup Examples](https://www.w3.org/TR/wai-aria-practices/examples/combobox/aria1.1pattern/listbox-combo.html) except I replaced input[type=text] with input[type=search]
```html
<label for="ex1-input"
       id="ex1-label"
       class="combobox-label">
  Choice 1 Fruit or Vegetable
</label>
<div class="combobox-wrapper">
  <div role="combobox"
       aria-expanded="false"
       aria-owns="ex1-listbox"
       aria-haspopup="listbox"
       id="ex1-combobox">
    <input type="search"
           aria-autocomplete="list"
           aria-controls="ex1-listbox"
           id="ex1-input">
  </div>
  <ul aria-labelledby="ex1-label"
      role="listbox"
      id="ex1-listbox"
      class="listbox hidden">
  </ul>
</div>
```

Actual: aXe audit fails with `Required ARIA child role not present: textbox` for the `#ex1-combobox` element.

Motivation: As per the [WAI-ARIA 1.1 recommendations for combobox](https://www.w3.org/TR/wai-aria-1.1/#combobox):
> Authors MUST ensure an element with role combobox contains or owns a text input element with role textbox or searchbox 

input[type=search] should satisfy this requirement.

Here is a minimal page containing the above snippet: https://srvpf.codesandbox.io/

axe-core version: 3.2.2-canary.2390925
axe-coconut extension version 3.8.1

**Comment by straker** [comment]
Thanks for reporting this, we'll take a look at the code to see what's going on.

**Comment by jeankaplansky** [comment]
Added a link to the rule help back to the ARIA documentation specific to role="combobox" which explains the intent of the spec.

**Comment by chandana7393** [comment]
Tested, working as expected.
![role_textbox](https://user-images.githubusercontent.com/45285854/67200473-7d9c3c80-f421-11e9-9bfb-9b9d8553effa.PNG)

**Tested Environment:**
Axe-coconut - 4.0.0.29404v
Chrome - 77.0.3865.120v
OS - Windows 10 64 bit.
